openpathsampling/sample.py

```diff
@@ -119,14 +121,15 @@ def __setitem__(self, key, value):
         self.append(value)
 
     def __eq__(self, other):
-        if len(self.samples) == len(other.samples):
-            for samp1 in self.samples:
-                if samp1 not in other.samples:
-                    return False
-
-            return True
-        else:
-            return False
+        return set(self.samples) == set(other.samples)
```

Comment:
Important, I checked the code again and it seems that we agreed before that one sample can only be present once. If you apply samples then the one with the same replica number will be replaced. So, at least by applying you cannot have the two samples with the same replica number present. Just to make sure I will change it to using a `collections.Counter` which will also compare the numbers of times a specific sample is present.